x/ccv/provider/keeper/relay.go

```diff
@@ -360,58 +358,40 @@ func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.Slas
 
 // EndBlockCCR contains the EndBlock logic needed for
 // the Consumer Chain Removal sub-protocol
+// TODO JEHAN: Stopping iteration here (dealt with by creating a "getFirst" function)
 func (k Keeper) EndBlockCCR(ctx sdk.Context) {
 	currentTime := ctx.BlockTime()
 	currentTimeUint64 := uint64(currentTime.UnixNano())
 
-	// iterate over initTimeoutTimestamps
-	var chainIdsToRemove []string
-	k.IterateInitTimeoutTimestamp(ctx, func(chainID string, ts uint64) (stop bool) {
-		if currentTimeUint64 > ts {
+	for _, initTimeoutTimestamp := range k.GetAllInitTimeoutTimestamps(ctx) {
+		if currentTimeUint64 > initTimeoutTimestamp.Timestamp {
 			// initTimeout expired
-			chainIdsToRemove = append(chainIdsToRemove, chainID)
-		}
-		// continue to iterate through all timed out consumers
-		return false
-	})
-	// remove consumers that timed out
-	for _, chainID := range chainIdsToRemove {
-		// stop the consumer chain and unlock the unbonding.
-		// Note that the CCV channel was not established,
-		// thus closeChan is irrelevant
-		err := k.StopConsumerChain(ctx, chainID, false)
-		if err != nil {
-			panic(fmt.Errorf("consumer chain failed to stop: %w", err))
+			// stop the consumer chain and unlock the unbonding.
+			// Note that the CCV channel was not established,
+			// thus closeChan is irrelevant
+			err := k.StopConsumerChain(ctx, initTimeoutTimestamp.ChainId, false)
+			if err != nil {
+				panic(fmt.Errorf("consumer chain failed to stop: %w", err))
+			}
 		}
 	}
 
-	// empty slice
-	chainIdsToRemove = nil
-
-	// Iterate over all consumers with established CCV channels and
-	// check if the first vscSendTimestamp in iterator + VscTimeoutPeriod
-	// exceed the current block time.
-	// Checking the first send timestamp for each chain is sufficient since
-	// timestamps are ordered by vsc ID.
-	k.IterateChannelToChain(ctx, func(ctx sdk.Context, _, chainID string) (stop bool) {
-		k.IterateVscSendTimestamps(ctx, chainID, func(_ uint64, ts time.Time) (stop bool) {
-			timeoutTimestamp := ts.Add(k.GetParams(ctx).VscTimeoutPeriod)
+	for _, channelToChain := range k.GetAllChannelToChains(ctx) {
+		// check if the first vscSendTimestamp in iterator + VscTimeoutPeriod
+		// exceed the current block time.
+		// Checking the first send timestamp for each chain is sufficient since
+		// timestamps are ordered by vsc ID.
+		vscSendTimestamp, found := k.GetFirstVscSendTimestamp(ctx, channelToChain.ChainId)
```

Comment:
We were previously stopping this iterator after one iteration. Made a GetFirst function instead